### PR TITLE
email/pw auth integration with firestore db

### PR DESCRIPTION
**Pull request by devintor**
barebones site but fully functional with email and password auth, error handling, routing, etc.

updates user info in firestore database

screen recording:
https://github.com/devintor/breadbox/assets/139282865/f324659b-7a96-484d-9219-9b49b79d8a16

### CODE REVIEW

**Comment by devintor** [comment]
Currently have everything related to GoogleAuth commented out because it has some bugs and I just wanted to show the email/password setup for now

File: backend/src/server.js
Comment (ecristerna) [inline backend/src/server.js:12]: which feature does this task/todos map to from the project plan?

File: backend/src/server.js
Comment (ecristerna) [inline backend/src/server.js:1]: let's find a better name for this file

File: frontend/src/components/Login.jsx
Comment (ecristerna) [inline frontend/src/components/Login.jsx:15]: rm

File: frontend/src/components/Login.jsx
Comment (ecristerna) [inline frontend/src/components/Login.jsx:21]: rm

File: frontend/src/components/Login.jsx
Comment (ecristerna) [inline frontend/src/components/Login.jsx:23]: is there a way to show the human readable string instead? 
Screenshot 2024-07-08 at 09 06 04

File: frontend/src/components/Login.jsx
Comment (ecristerna) [inline frontend/src/components/Login.jsx:63]: rm

File: frontend/src/components/Login.jsx
Comment (ecristerna) [inline frontend/src/components/Login.jsx:60]: rename this class?

File: frontend/src/components/Profile.jsx
Comment (ecristerna) [inline frontend/src/components/Profile.jsx:49]: rm

File: frontend/src/components/Register.jsx
Comment (ecristerna) [inline frontend/src/components/Register.jsx:26]: can you explain this process? is `currentUser` the newly created user, or an existing one? then what are we doing with the `setDoc()` call?

**Review by ecristerna — APPROVED** [review]
Let's remove all the commented code, and file that are not needed/used today, before merging this to main. If you want to keep the Google OAuth code, you can fork this branch and create a new one, but do not merge code and files (and images) into main that are not actually used.

Also, please make sure there are no `console.log` in your code before submitting.

**Pull request merged**
